pkgs/tools/networking/curl/default.nix

```diff
@@ -66,6 +66,7 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./7.79.1-darwin-no-systemconfiguration.patch
+    ./7.82.0-openssl-fix-CN-check.patch
```

Comment:
This seems to be mostly the [exact same patch as upstream](https://patch-diff.githubusercontent.com/raw/curl/curl/pull/8560.patch). If possible, I'd recommend `fetchpatch` instead of adding yet another file on the repo, if it's accessible online.

Reply:
gador [replied](https://github.com/NixOS/nixpkgs/pull/167993#pullrequestreview-937132247):

> Not possible. Curl needs to be build for fetchpatch. This will cause an infinite recusion.
> There is a comment about that above the patch line.

For those looking for that comment, it's (_way_ above) on lines 29–32 of `pkgs/tools/networking/curl/default.nix`.

> This is correct as it is.

:+1: